aiohttp/connector.py

```diff
@@ -770,6 +774,9 @@ def __init__(
         self._throttle_dns_events: Dict[Tuple[str, int], EventResultOrError] = {}
         self._family = family
         self._local_addr = local_addr
+        self._network_interface = network_interface
+        if self._network_interface and not hasattr(socket, "SO_BINDTODEVICE"):
+            raise OSError("Binding to interface is not supported by your OS.")
```

Comment:
Isn't it more of a `RuntimeError` for us? Or rather its subclass `NotImplementedError`? `OSError` tends to represent low-level syscall interactions, while the current check is more of an assertion happening closer to the application level.
@Dreamsorcerer WDYT?

Reply:
You are right.

> This exception is raised when a system function returns a system-related error.

We could also make our own exception, but depends on what you see fit.

Reply:
A custom exception is probably an overkill here.